src/main/scala/org/broadinstitute/dsde/workbench/avram/api/CollectionsServlet.scala

```diff
@@ -1,39 +1,28 @@
 package org.broadinstitute.dsde.workbench.avram.api
 
-import io.circe.generic.auto._
-import io.circe.syntax._
 import java.util.UUID
-import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
 
-import org.broadinstitute.dsde.workbench.avram.model.{AvramException, SamResource}
+import io.circe.generic.auto._
+import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
+import org.broadinstitute.dsde.workbench.avram.model.SamResource
 import org.broadinstitute.dsde.workbench.avram.service.CollectionsService
 
 
 class CollectionsServlet extends HttpServlet with AvramServlet {
 
   private val collectionsService = new CollectionsService()
 
-  override def doGet(request: HttpServletRequest, response: HttpServletResponse) = {
-    val collection = handleAuthenticatedRequest(request, response) { userInfo =>
-      response.setContentType("application/json")
-      val parameter = request.getPathInfo.stripPrefix("/")
-      val collectionExternalId = UUID.fromString(parameter)
-      val future = collectionsService.getCollection(collectionExternalId)
-      registerOnComplete(future, request, response)
+  override def doGet(request: HttpServletRequest, response: HttpServletResponse): Unit = {
+    handleAuthenticatedRequest(request, response) { _ =>
+      val collectionExternalId = UUID.fromString(request.getPathInfo.stripPrefix("/"))
```

Comment:
It seems like this line is also something that could fail (e.g. if the request doesn't have a valid UUID in the path) so it should be something that gets lifted to an `Either`.

Reply:
Yeah, I wasn't being too picky about that right now because I think we're going to want to use Jersey or something like that which will change this code. You're right to point it out, though, because the code should be complete in the current context so I'll address this.

Reply:
Also, I want as little code in here as possible. I'm on the fence as to whether to return the 400 from here or pass an `Option[String]` to `collectionService.getCollection` and have the input validation there. I guess here makes more sense.